content/essentials/components-and-props.mdx

```diff
@@ -0,0 +1,360 @@
+// App.ts
+import { MyComponent } from './MyComponent';
+
+function App() {
+  return (
+    <div>
+      <MyComponent />
+    </div>
+  );
+}
+```
+
+**Default import:**
+
+```typescript
+// App.ts
+import MyComponent from './MyComponent';
+
+function App() {
+  return (
+    <div>
+      <MyComponent />
+    </div>
+  );
+}
+```
+
+### Importing Solid and its utilities
+
+For a component to have reactivity and the other features provided by Solid, you will often need to import necessary functions and utilities.
+
+```typescript
+import { createSignal, Show } from 'solid-js';
+```
+
+## Component props
+
+Props are a way through which data is passed from one component to another. 
+In Solid, props are used to pass data from a parent component to a child component, similar to how function arguments work. 
+
+```typescript
+function App() {
+  // Passing a prop "name" with the value "Ryan Carniato" to MyComponent
+  return (
+    <div>
+      <MyComponent name="Ryan Carniato" />
+    </div>
+  );
+}
+
+function MyComponent(props) {
+  // Accessing the "name" prop within MyComponent
+  return <div>Hello {props.name}</div>;
+}
+```
+
+### `mergeProps`
+
+In Solid, [`mergeProps`](/reference/mergeProps) is a utility function designed to merge multiple potentially reactive objects together. 
+Its behavior is similar to [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) but with a key distinction: it retains the reactivity of the properties being merged.
+This ensures that when individual properties within the merged object change, their reactivity is not lost.
+
+A common use case for `mergeProps` is when you want to provide default props:
+
+```typescript
+import { mergeProps } from 'solid-js';
+
+function MyComponent(props) {
+  // Using mergeProps to set default values for props
+  const finalProps = mergeProps({ defaultName: "Ryan Carniato" }, props);
+  
+  return <div>Hello {finalProps.defaultName}</div>;
+}
+
+// Usage: <MyComponent defaultName="Ryan Carniato" />
+```
+
+In the example above, if no `defaultName` prop is provided when using `<MyComponent />`, it will default to `"Ryan Carniato"` because of `mergeProps`. 
+If a value is provided, it will be used instead, all while retaining the reactivity of the `defaultName` prop.
+
+### Destructuring props
+
+Props objects are typically readonly, ensuring that a child component does not directly modify the data belonging to its parent. 
+This encourages one-way data flow, a pattern often seen to promote more predictable data management. 
+Since the properties within the props object are reactive, special care is needed when working with them.
+
+When destructuring props directly, you run the risk of breaking this reactivity, leading to potential unexpected behaviors in your components:
+
+```typescript
+function MyComponent(props) {
+  const name = props.name // ❌: breaks reactivity and will not update when the prop value changes
```

Comment:
```suggestion
  const name = () => props.name // ❌: breaks reactivity and will not update when the prop value changes
```

Reply:
I'm proposing this as a way to not break reactivity, which would be the line below

Reply:
OH got it thank you